app/src/main/java/com/aptoide/uploader/UploaderApplication.java

```diff
@@ -90,6 +97,16 @@
     startFlurryAgent();
     initializeRakam();
     getUploadManager().start();
+    refreshInstalledApps();
+  }
+
+  private void refreshInstalledApps() {
+    getInstallManager();
```

Comment:
The ideia is to only create it the InstallManager if it wasnt yet created, like a singleton, but if it null, its also initializing the RoomInstalledPersistence since we need it to initialize the InstallManager. 
I removed this line and changed the line after this like suggested, but I since the method its initializing two variables and the installManager variable will only be null if it wasnt created yet I am usure how to improve this with an inline

Reply:
The idea behind, lazy initialization, is correct. The issue I see is, without looking into the method, this looks like a simple get method but you are not assigning the variable anywhere. It is a method with side effects: you are assigning the return inside to a global variable. If this was a method like: initInstallManager() it would be better, but not good.

My suggestion would be replacing all installManager variable usages by the method getInstallManager()